apps/frontend/src/pages/chat/chat.tsx

```diff
@@ -1,21 +1,69 @@
-import { QuizCategoriesForm } from "~/libs/components/components.js";
-import { useCallback } from "~/libs/hooks/hooks.js";
+import { DataStatus } from "~/libs/enums/enums.js";
+import {
+	useAppDispatch,
+	useAppSelector,
+	useEffect,
+	useRef,
+} from "~/libs/hooks/hooks.js";
+import { actions as chatActions } from "~/modules/chat/chat.js";
+import { actions as quizActions } from "~/modules/quiz/quiz.js";
 
+import {
+	ButtonsController,
+	ChatMessage,
+	InitialMessages,
+	MessageLoader,
+} from "./libs/components/components.js";
 import styles from "./styles.module.css";
 
 const Chat: React.FC = () => {
-	const handleCategoriesSubmit = useCallback(() => {
-		// TODO: Add a request to Backend
-	}, []);
+	const dispatch = useAppDispatch();
+	const chatEnd = useRef<HTMLDivElement | null>(null);
+
+	const { messages, messageStatus, threadId } = useAppSelector((state) => ({
+		messages: state.chat.messages,
+		messageStatus: state.chat.dataStatus,
+		quizCategories: state.categories.items,
+		scores: state.quiz.scores,
+		selectedCategories: state.chat.selectedCategories,
+		threadId: state.chat.threadId,
+	}));
+
+	const scrollToBottom = (): void => {
+		if (chatEnd.current) {
+			chatEnd.current.scrollIntoView({ behavior: "smooth" });
+		}
+	};
+
+	useEffect(() => {
+		scrollToBottom();
+	}, [messages]);
+
+	useEffect(() => {
+		void dispatch(quizActions.getScores());
+		void dispatch(chatActions.initConversation());
+	}, [dispatch]);
 
 	return (
 		<main className={styles["page-container"]}>
-			<section className={styles["quiz-form-container"]}>
-				<QuizCategoriesForm
-					buttonLabel="Select"
-					onSubmit={handleCategoriesSubmit}
-				/>
-			</section>
+			<div>
+				<ul className={styles["container"]}>
+					{threadId && <InitialMessages />}
+					{messages.map((message, index) => {
+						return (
+							<ChatMessage
+								author={message.author}
+								key={index}
+								payload={message.payload}
+								type={message.type}
+							/>
+						);
+					})}
+					{threadId && <ButtonsController />}
```

Comment:
This can be redone by having the "none" value for buttonsMode property of initial slice and setting to needed button group when threadId is fetched (initializeChat thunk). That way we won't have this bracket expression here